opm/material/thermal/EclSpecrockLawParams.hpp

```diff
@@ -62,7 +62,7 @@ class EclSpecrockLawParams : public EnsureFinalized
         // integrate the heat capacity to compute the internal energy
         Scalar curU = temperature[0]*heatCapacity[0];
         std::vector<Scalar> T(temperature.size());
-        std::vector<Scalar> u(heatCapacity.size());
+        std::vector<Scalar> u(temperature.size());
         for (unsigned i = 0; i < temperature.size(); ++ i) {
```

Comment:
While the variable renaming is fine I have some comments about the code:
 - This loop would probably have been clearer if written to loop from 1 rather than 0 (the `break` could have been avoided).
 - There may be a bug in the math on line 80: I the integral of the linear heat capacity polynomial should include the term `m*(-T0*T1 + T0*T0)`, from integrating the constant term of the polynomial. That in turn should let you simplify the answer to `deltaU = 0.5*(c_v0 + c_v1)*(T1 - T0)` which is consistent with the midpoint rule.

Reply:
I don't really get what you mean: the formula is the exact expression for the integral of c_v from T_0 to T_1 if c_v is assumed to be a linear function. at least that is what it supposed to be.

Reply:
> the formula is the exact expression for the integral of c_v from T_0 to T_1 if c_v is assumed to be a linear function. at least that is what it supposed to be.

The integral of a linear function is exactly reproduced by the midpoint formula, which is not what you have here, so I think it is a bug.

Reply:
I don't think it is incorrect:

```
\int_T0^T1 mT + c dT = 0.5 *m T1^2 + c*T1 - (0.5*m*T0^2 + c*T0) =  0.5*m*(T1^2 - T0^2) + c*(T1 - T0)
```
and that's exactly what is implemented?

Reply:
The mistake is this, I believe: while the above is correct for a linear function with slope `m` and constant term `c`, your constant term is wrong. With the formula
```
y = m T + c
```
then y is equal to c when T = 0. However you use `c_v0` for the constant, and y should not be equal to c at T= 0 but at T = T0. If you want to use `c_v0` for the constant you should rather integrate
```
y = m (T - T0) + c_v0
```